dataRetriever/blockchain/baseBlockchain.go

```diff
@@ -70,3 +71,25 @@ func (bbc *baseBlockChain) SetCurrentBlockHeaderHash(hash []byte) {
 	bbc.currentBlockHeaderHash = hash
 	bbc.mut.Unlock()
 }
+
+func (bbc *baseBlockChain) SetHighestFinalBlockCoordinates(nonce uint64, headerHash []byte, rootHash []byte) {
+	bbc.mut.Lock()
+
+	bbc.finalCoordinates.blockNonce = nonce
+	bbc.finalCoordinates.blockHash = headerHash
+	bbc.finalCoordinates.blockRootHash = rootHash
+
+	bbc.mut.Unlock()
+}
+
+func (bbc *baseBlockChain) GetHighestFinalBlockCoordinates() (nonce uint64, hash []byte, rootHash []byte) {
```

Comment:
missing comments.
Let's find another way to return those []byte slices as that is dangerous (we can swap them by mistake where we call this function)
Also, we stopped using named output parameters. The code will be prone to future errors.

Reply:
Added comments. As discussed, left the signature of the function as it was.